src/JsonSchema/Validator.php

```diff
@@ -41,14 +41,11 @@ class Validator extends BaseConstraint
      *
      * @param mixed $value
      * @param mixed $schema
-     * @param int   $checkMode
-     *
-     * @return int
      *
      * @phpstan-param int-mask-of<Constraint::CHECK_MODE_*> $checkMode
      * @phpstan-return int-mask-of<Validator::ERROR_*>
      */
-    public function validate(&$value, $schema = null, $checkMode = null)
+    public function validate(&$value, $schema = null, int $checkMode = null): int
```

Comment:
```suggestion
    public function validate(&$value, $schema = null, ?int $checkMode = null): int
```
Implicit nullable parameter types are deprecated in PHP 8.4 (see https://wiki.php.net/rfc/deprecate-implicitly-nullable-types)

Reply:
Great spot and there should have been a pickup by the GitHub actions. I'll investigate 🔍